packages/dara-core/tests/python/test_variable.py

```diff
@@ -70,3 +77,31 @@ def test_derived_variable_uid_run_as_task(self):
 
         with self.assertRaises(ValueError):
             DerivedVariable(test_resolver, variables=[variable], uid='test_task_var'),
+
+async def test_derived_variables_with_df_nan():
```

Comment:
You could have kept it as a py_component test, it was just about not calling `to_string` on the DF directly so we test the serialization. This works just as well though :)